artifact/tar/platform/libarchive/wrapper.hpp

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_LIB_ARCHIVE_WRAPPER_HPP
+#define MENDER_LIB_ARCHIVE_WRAPPER_HPP
+
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+#include <archive.h>
+#include <archive_entry.h>
+#include <sys/types.h>
+
+#include <common/io.hpp>
+#include <common/log.hpp>
+#include <common/error.hpp>
+#include <common/expected.hpp>
+
+namespace mender {
+namespace libarchive {
+namespace wrapper {
+
+using namespace std;
+
+namespace error = mender::common::error;
+namespace expected = mender::common::expected;
+namespace io = mender::common::io;
+using mender::common::error::Error;
+
+namespace log = mender::common::log;
+
+class ReaderContainer {
```

Comment:
I  believe this could be just a passive `struct` (with public fields) defined in `wrapper.cpp`, it's not used anywhere else, right?

Reply:
mby, but does it really make a difference?

Reply:
Hmm, kk. If it's the way it's usually done. 

It hides less data though, but have at it.